### PR TITLE
Add CI options to prepare and merge release branches

### PR DESCRIPTION
**Pull request by uurien**
### What does this PR do?

Adds two Github action jobs to be executed manually. One to prepare release proposal PR, and other to rebase the branch in release branches.

### Motivation

Try to make releases easier and adds the possibility to block "rebase merge" option from merge options. To prevent rebasing to master. Eventually we will be able to prevent manual merges in `v[\d].x` branches.

### CODE REVIEW

**Review by rochdev — COMMENTED** [review]
I wonder if this could also be run as a nightly job that automatically releases dev versions, or even as a daily job that releases whatever has been merged in the last 24h.

File: .github/workflows/prepare-release-proposal.yml
Comment (uurien) [reply .github/workflows/prepare-release-proposal.yml:9]: True, we could remove this option and execute always in both.

File: .github/workflows/prepare-release-proposal.yml
Comment (uurien) [reply .github/workflows/prepare-release-proposal.yml:16]: Eventually, we might want to release a patch version with more minor commits in `master`, don't we?

File: .github/workflows/prepare-release-proposal.yml
Comment (uurien) [reply .github/workflows/prepare-release-proposal.yml:27]: Maybe most of the steps, but we need to have `Create PR` step in a separate step before the package.json update, because it needs different GH access token

File: .github/workflows/rebase-release-proposal.yml
Comment (uurien) [reply .github/workflows/rebase-release-proposal.yml:4]: Sometimes happen that the release proposal branch is green and have approval, but we are waiting because we want more approvals or some release notes updates.
I think that this should be manual for now, because once that this is merged, we should click on "publish npm" button and create a release in GH.

File: .github/workflows/prepare-release-proposal.yml
Comment (rochdev) [reply .github/workflows/prepare-release-proposal.yml:9]: Why would we want to do that? Usually we release either minor or patch depending on what new commits are on the main branch. I don't see why even a hot fix couldn't be a minor if it includes minor changes.

I guess my main argument against having this as an option is that you end up having to go through the commits to figure this out, which is part of the work this would otherwise remove.

File: .github/workflows/prepare-release-proposal.yml
Comment (rochdev) [reply .github/workflows/prepare-release-proposal.yml:25]: I guess I meant more for whatever can be done locally, like checking out the right release line, creating the proposal branch with the right name, merging commits, committing the new version, etc.

File: .github/workflows/rebase-release-proposal.yml
Comment (rochdev) [reply .github/workflows/rebase-release-proposal.yml:4]: I still think this could be automated even if reviews are required so that it would wait for the reviews, but that may be out of scope of this PR and could be re-evaluated later after we have evaluated the manual steps for a little while.

File: .github/workflows/prepare-release-proposal.yml
Comment (rochdev) [reply .github/workflows/prepare-release-proposal.yml:25]: Most of this is already extracted to a script, so that's fine for now and definitely not a blocker.

File: .github/workflows/prepare-release-proposal.yml
Comment (rochdev) [reply .github/workflows/prepare-release-proposal.yml:9]: The rest of the PR LGTM, so this would be the only thing I'd ideally like to see addressed. I'll still approve the PR so that if you prefer to make this improvement as a second step you're not blocked.

**Review by rochdev — APPROVED** [review]

File: .github/workflows/prepare-release-proposal.yml
Comment (uurien) [reply .github/workflows/prepare-release-proposal.yml:9]: I'll try to do some tests in other repo to see if this is easily doable, if it is I'll address it in this PR.

Comment (uurien) [reply .github/workflows/prepare-release-proposal.yml]: Changed

**Review by rochdev — APPROVED** [review]

**Pull request merged**
